src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Overview/ThorchainSaversOverview.tsx

```diff
@@ -118,28 +114,42 @@ export const ThorchainSaversOverview: React.FC<OverviewProps> = ({
     [accountId, assetNamespace, assetReference, chainId, defaultAccountId, highestBalanceAccountId],
   )
 
-  const opportunityData = useAppSelector(state =>
+  const earnOpportunityData = useAppSelector(state =>
     selectEarnUserStakingOpportunityByUserStakingId(state, opportunityDataFilter),
   )
 
+  const opportunitiesMetadata = useAppSelector(state => selectStakingOpportunitiesById(state))
+
+  const opportunityMetadata = useMemo(
+    () => opportunitiesMetadata[assetId as StakingId],
+    [assetId, opportunitiesMetadata],
+  ) as OpportunityMetadata<DefiProvider.ThorchainSavers, DefiType.Staking> | undefined
```

Comment:
Needed since indexing `opportunitiesMetadata[assetId as StakingId]` will give us `OpportunityMetadata` with default generics - we could potentially have a new `selectStakingOpportunityById` selector with `<T, U>` generics to avoid casting?